File: packages/zodiac-dart/lib/src/models/chinese_animal.dart
/// Chinese Zodiac animal model.
class ChineseAnimal {
  final String id;
  final String name;
  final String symbol;
  final int order;
  final String yinYang; // 'yin' | 'yang'
  final String fixedElement;
  final String description;
  final List<String> keywords;
  final List<String> strengths;
  final List<String> weaknesses;
  final List<String> compatible;
  final String iconId;

  const ChineseAnimal({
    required this.id,
    required this.name,
    required this.symbol,
    required this.order,
    required this.yinYang,
    required this.fixedElement,
    required this.description,
    required this.keywords,
    required this.strengths,
    required this.weaknesses,
    required this.compatible,
    required this.iconId,
  });
}

File: packages/zodiac-js/src/models/chinese-animal.ts
/**
 * Chinese Zodiac animal model.
 *
 * Represents one of the 12 Chinese Zodiac animals with all metadata.
 */
export interface ChineseAnimal {
  /** Stable, language-independent identifier (e.g., "rat") */
  readonly id: string;

  /** English display name */
  readonly name: string;

  /** Emoji or symbol */
  readonly symbol: string;

  /** Position in the 12-animal cycle (1–12) */
  readonly order: number;

  /** Yin/Yang classification */
  readonly yinYang: 'yin' | 'yang';

  /** Fixed element associated with this animal */
  readonly fixedElement: string;

  /** Traditional description */
  readonly description: string;

  /** Characteristic keywords */
  readonly keywords: readonly string[];

  /** Traditional strengths */
  readonly strengths: readonly string[];

  /** Traditional weaknesses */
  readonly weaknesses: readonly string[];

  /** Compatible animal IDs */
  readonly compatible: readonly string[];

  /** Icon identifier */
  readonly iconId: string;
}
